### DISCUSSION TITLE
how to handle @angular-devkit/build-angular?

### DISCUSSION THREAD

**Discussion by japj** [renovatebot/renovate#8594] [Request Help] [answered]
Hello,

I am using the default base config and am "updating angularcli monorepo"

I have the following devDependencies in the renovate commit:
from:
```
    "@angular-devkit/build-angular": "0.901.0",
    "@angular-devkit/build-ng-packagr": "0.901.0",
    "@angular/cli": "9.1.0",
```

to:
```
    "@angular-devkit/build-angular": "0.1101.4",
    "@angular-devkit/build-ng-packagr": "0.1002.0",
    "@angular/cli": "9.1.14",
```
(this is for an angular 9.x application)

In this specific case it seems that for angular 9 it should have been:

```
"@angular-devkit/build-angular": "0.901.14",
"@angular-devkit/build-ng-packagr": "0.901.14",
```

Note the very weird version numbering (starting with 0.*).

Is there a way to properly handle this?

**Comment by rarkins** [comment, marked as answer]
This might work:

```json
{
  "packageRules": [{
    "packagePatterns": ["^angular-devkit"],
    "allowedVersions": "^0\\.9"
  }]
}
```

I'm not sure we have any way to today to make one package's versions automatically be based on another's, so this rule would need to be manual.

**Reply by japj** [reply]
Thanks for the config settings, I believe there might be a typo though, because of the missing @ before angular-devkit (testing now if that fixes the issue).

Since I am very new to renovate, it is still a bit tricky to figure out what should work "out-of-the-box" versus what is more "tuning" for a specific repository/scenario.

However, I am very happy with how it is working so far!

**Reply by japj** [reply]
unfortunately I get an error:

"validationMessage": "The following allowedVersions does not parse as a valid version or range: \"^0\\\\.9\"",

**Reply by rarkins** [reply]
Something as whacky as the angular dev kit versioning is unlike to ever work out of the box, but many other things do. 


Try this corrected version:

```
{
  "packageRules": [{
    "packagePatterns": ["^@angular-devkit"],
    "allowedVersions": "/^0\\.9/"
  }]
}
```
